apps/precursor-chemicals/sections/summary-data-sections.js

```diff
@@ -0,0 +1,156 @@
+        field: 'licence-holder-details',
+        parse: (list, req) => {
+          const licenseHolderDetails = [];
+          licenseHolderDetails.push(req.sessionModel.get('company-name'));
+          if (req.sessionModel.get('company-number')) {
+            licenseHolderDetails.push(req.sessionModel.get('company-number').toUpperCase());
+          }
+          if (req.sessionModel.get('website-url')) {
+            licenseHolderDetails.push(req.sessionModel.get('website-url'));
+          }
+          licenseHolderDetails.push(req.sessionModel.get('telephone'));
+          licenseHolderDetails.push(req.sessionModel.get('email'));
+          return licenseHolderDetails.join('\n');
+        }
+      },
+      {
+        step: '/licence-holder-address',
+        field: 'licence-holder-address',
+        parse: (list, req) => {
+          const licenceHolderAddress = [];
+          licenceHolderAddress.push(req.sessionModel.get('licence-holder-address-line-1'));
+          if (req.sessionModel.get('licence-holder-address-line-2')) {
+            licenceHolderAddress.push(req.sessionModel.get('licence-holder-address-line-2'));
+          }
+          licenceHolderAddress.push(req.sessionModel.get('licence-holder-town-or-city'));
+          licenceHolderAddress.push(req.sessionModel.get('licence-holder-postcode'));
+          return licenceHolderAddress.join('\n');
+        }
+      },
+      {
+        step: '/premises-address',
+        field: 'premises-address-details',
+        parse: (list, req) => {
+          if (!req.sessionModel.get('steps').includes('/premises-address')) {
+            return null;
+          }
+          const addressDetails = [];
+          addressDetails.push(req.sessionModel.get('premises-address-line-1'));
+          if (req.sessionModel.get('premises-address-line-2')) {
+            addressDetails.push(req.sessionModel.get('premises-address-line-2'));
+          }
+          addressDetails.push(req.sessionModel.get('premises-town-or-city'));
+          addressDetails.push(req.sessionModel.get('premises-postcode'));
+          return addressDetails.join('\n');
+        }
+      },
+      {
+        step: '/reuse-premises-address',
+        field: 'is-premises-address-same'
+      },
+      {
+        step: '/premises-contact-details',
+        field: 'premises-contact-details',
+        parse: (list, req) => {
+          const contactDetails = [];
+          contactDetails.push(req.sessionModel.get('premises-telephone'));
+          contactDetails.push(req.sessionModel.get('premises-email'));
+          return contactDetails.join('\n');
+        }
+      },
+      {
+        step: '/responsible-officer-details',
+        field: 'responsible-officer-details',
+        parse: (val, req) => {
+          const formattedResponsibleOfficerDetails = Array(
+            req.sessionModel.get('responsible-officer-fullname'),
+            req.sessionModel.get('responsible-officer-email')
+          ).join('\n');
```

Comment:
We have two formats in this file for uniting several fields in an array and joining them for display in the summary. No problem as both will work, but we may want to decide on one and refactor for general readability/continuity.

Reply:
agree, I would personally prefer this option for readability:

```
const myArray = [
  req.sessionModel.get('value1'),
  req.sessionModel.get('value2'),
];

return myArray.join('\n');
```

Reply:
Resolved by PR #21